packages/shared/src/components/sources/SourceActions/SourceActionsFollow.tsx

```diff
@@ -12,15 +13,32 @@ interface SourceActionsFollowProps {
 
 const SourceActionsFollow = (props: SourceActionsFollowProps): ReactElement => {
   const { className, isSubscribed, isFetching, onClick, variant } = props;
-  const label = isSubscribed ? 'Unfollow' : 'Follow';
+  const [isHovered, setIsHovered] = useState(false);
+
+  let label = isSubscribed ? 'Following' : 'Follow';
+
+  if (isHovered && isSubscribed) {
+    label = 'Unfollow';
+  }
+
   return (
```

Comment:
This one couldn't be the generic FollowButton now?
(not sure why it would be different?)

Reply:
I guess because we only allow it for users for now :)

Reply:
yes, I used only base components and we will transition to `FollowButton` in phase 2 & 3